src/components/datetime/DateTimeController.cpp

```diff
@@ -75,6 +75,22 @@ void DateTime::UpdateTime(uint32_t systickCounter) {
   minute = time.minutes().count();
   second = time.seconds().count();
 
+  if (minute == 0 and not isHourAlreadyNotified) {
+    isHourAlreadyNotified = true;
+    if (systemTask != nullptr)
+      systemTask->PushMessage(System::Messages::OnNewHour);
+  } else if (minute != 0) {
+    isHourAlreadyNotified = false;
+  }
+
+  if ((minute == 0 or minute == 30) and not isHalfHourAlreadyNotified) {
+    isHalfHourAlreadyNotified = true;
+    if (systemTask != nullptr)
+      systemTask->PushMessage(System::Messages::OnNewHalfHour);
```

Comment:
Lack of braces is inconsistent with the rest of the code.

Reply:
Now MD is causing me issues.  Is this what is required (from a braces point of view) to resolve please?

   if (systemTask != nullptr) {  
       systemTask->PushMessage(System::Messages::OnNewHalfHour);  
    }

Reply:
Yep, just adding the two, here and other places where they're missing.